src/git_changelog/cli.py

```diff
@@ -468,30 +570,39 @@ def main(args: list[str] | None = None) -> int:
     parser = get_parser()
     opts = parser.parse_args(args=args)
 
-    if opts.release_notes:
+    # Determine which arguments were explicitly set with the CLI
```

Comment:
Since you removed the defaults from the arguments, isn't it enough to check if an argument value is `None` (unset) or something else (set)? Or maybe we have specific logic for `None` for some of them? In that case yes the sentinel is needed.

Reply:
Boolean switches are False always and for some options, `None` is an appropriate value (e.g., sections).

It felt safer to have a clear understanding of what the user actually set, and at the same time, initialize the settings with the content from the config file.

Reply:
Ah, yes, we do use `action="store_true"`. Good points. OK for the sentinels then, thanks for the clarification!